.github/workflows/pull-request.yml

```
@@ -0,0 +1,272 @@
+      - name: Run formatting if check fails
+        if: failure()
+        run: npm run format
+
+      - name: Check for type errors
+        if: steps.changed-files.outputs.only_changed != 'true'
+        run: npm run typecheck
+
+      - name: Check for linting errors in modified files
+        if: steps.changed-files.outputs.only_changed != 'true'
+        env: 
+          CHANGED_FILES: ${{ steps.changed_files.outputs.all_changed_files }}
```

actionlint (1.7.4)

property "changed_files" is not defined in object type {changed-files: {conclusion: string; outcome: string; outputs: {string => string}}}

(expression)